snap/container.go

```diff
@@ -95,3 +97,184 @@ func Open(path string) (Container, error) {
 
 	return nil, fmt.Errorf("cannot open snap: unknown header: %q", header)
 }
+
+var (
+	// ErrBadModes is returned by ValidateContainer when the container has files with the wrong file modes for their role
+	ErrBadModes = errors.New("snap is unusable due to bad permissions; contact developer")
+	// ErrMissingPaths is returned by ValidateContainer when the container is missing required files or directories
+	ErrMissingPaths = errors.New("snap is unusable due to missing files; contact developer")
```

Comment:
`snap pack` is likely to be used by a developer and since the errors are also returned as-is there, maybe consider dropping the `; contact developer` suffix and only append it at the `check_snap.go` call site.